### ISSUE TITLE
Support for Project Configuration Files

### ISSUE THREAD

**Issue by catalys1** [catalys1/dnnutil#4]
I'm envisioning the ability to offload many of the details of a specific model run into a configuration file. The idea is that the configuration file could allow the same code to be run on a lot of different configurations of data, network, hyperparameters, etc., by just using a different config file. The configuration file could contain things like

- batch size, learning rate, etc
- parameters used in creating the network
- parameters used in setting up the dataloader
- specifics about the optimizer
- or really anything that you would like to be able to change up as your working on developing a new working model

Using a configuration file would be useful for several reasons

1. You could run multiple different configurations without having to change the code
2. You would avoid having to specify a whole lot of commandline arguments
3. The configuration file could be saved in the directory of a specific run, documenting exactly how that run was performed without having to rely on additional notes provided at run time or after the fact

The configuration file could probably be something as simple as a json file with certain fields. It could also be extensible, in that the programmer can add any fields he wants as long as he sets his code up to use them correctly.

**Comment by catalys1** [comment]
I've written a working version of this, it is currently in the config branch.

**Comment by catalys1** [comment]
Merged this functionality into master https://github.com/catalys1/dnnutil/commit/19a2641e34ac1267a237f9ff0d637a34b0b9fd47
